File: features/notes/src/main/java/com/nivasafinance/features/notes/service/NotesService.java
package com.nivasafinance.features.notes.service;

import com.nivasafinance.common.base.model.PaginatedResponse;
import com.nivasafinance.common.base.model.PaginationRequest;
import com.nivasafinance.features.notes.dto.NotesRequest;
import com.nivasafinance.features.notes.dto.NotesResponse;
import com.nivasafinance.features.notes.dto.NotesUpdateRequest;
import java.util.UUID;

public interface NotesService {
    NotesResponse createNotes(NotesRequest notesRequest);
    NotesResponse updateNotes(UUID notesId, NotesUpdateRequest notesUpdateRequest);
    NotesResponse patchNotes(UUID notesId, NotesUpdateRequest notesUpdateRequest);
    void deleteNotes(UUID notesId);
    NotesResponse getNotesById(UUID notesId);
    PaginatedResponse<NotesResponse> getAllNotes(PaginationRequest paginationRequest);
}

File: features/notes/src/main/java/com/nivasafinance/features/notes/service/NotesService.kt
package com.nivasafinance.features.notes.service

import base.model.PaginatedResponse
import base.model.PaginationRequest
import com.nivasafinance.features.notes.dto.NotesRequest
import com.nivasafinance.features.notes.dto.NotesResponse
import com.nivasafinance.features.notes.dto.NotesUpdateRequest
import java.util.UUID

interface NotesService {
    fun createNotes(notesRequest: NotesRequest): NotesResponse
    fun updateNotes(notesId: UUID, notesUpdateRequest: NotesUpdateRequest): NotesResponse
    fun patchNotes(notesId: UUID, notesUpdateRequest: NotesUpdateRequest): NotesResponse
    fun deleteNotes(notesId: UUID)
    fun getNotesById(notesId: UUID): NotesResponse
    fun getAllNotes(paginationRequest: PaginationRequest): PaginatedResponse<NotesResponse>
}       